examples/manifold/plot_lle_digits.py

```diff
@@ -3,246 +3,181 @@
+#   generated by PCA in this example. It ensures global stability  of the embedding,
+#   i.e., the embedding does not depend on random initialization.
+from sklearn.decomposition import TruncatedSVD
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.ensemble import RandomTreesEmbedding
+from sklearn.manifold import (
+    Isomap,
+    LocallyLinearEmbedding,
+    MDS,
+    SpectralEmbedding,
+    TSNE,
+)
+from sklearn.neighbors import NeighborhoodComponentsAnalysis
+from sklearn.pipeline import make_pipeline
+from sklearn.random_projection import SparseRandomProjection
+
+embeddings = {
+    "Random projection embedding": SparseRandomProjection(
+        n_components=2, random_state=42
+    ),
+    "Truncated SVD embedding": TruncatedSVD(n_components=2),
+    "Linear Discriminant Analysis embedding": LinearDiscriminantAnalysis(
+        n_components=2
+    ),
+    "Isomap embedding": Isomap(n_neighbors=n_neighbors, n_components=2),
+    "Standard LLE embedding": LocallyLinearEmbedding(
+        n_neighbors=n_neighbors, n_components=2, method="standard"
+    ),
+    "Modified LLE embedding": LocallyLinearEmbedding(
+        n_neighbors=n_neighbors, n_components=2, method="modified"
+    ),
+    "Hessian LLE embedding": LocallyLinearEmbedding(
+        n_neighbors=n_neighbors, n_components=2, method="hessian"
+    ),
+    "LTSA LLE embedding": LocallyLinearEmbedding(
+        n_neighbors=n_neighbors, n_components=2, method="ltsa"
+    ),
+    "MDS embedding": MDS(n_components=2, n_init=1, max_iter=100),
+    "Random Trees embedding": make_pipeline(
+        RandomTreesEmbedding(n_estimators=200, max_depth=5, random_state=0),
+        TruncatedSVD(n_components=2),
+    ),
+    "Spectral embedding": SpectralEmbedding(
+        n_components=2, random_state=0, eigen_solver="arpack"
+    ),
+    "t-SNE embeedding": TSNE(
+        n_components=2, init="pca", learning_rate="auto", random_state=0
+    ),
+    "NCA embedding": NeighborhoodComponentsAnalysis(
+        n_components=2, init="random", random_state=0
+    ),
+}
+
+# %%
+# Once we declared all the methodes of interest, we can run and perform the projection
+# of the original data. We will store the projected data as well as the computational
+# time needed to perform each projection.
+from time import time
+
+projections, timing = {}, {}
+for name, transformer in embeddings.items():
+    if name.startswith("Linear Discriminant Analysis"):
+        data = X.copy()
+        data.flat[:: X.shape[1] + 1] += 0.01  # Make X invertible
+    else:
+        data = X
+
+    print(f"Computing {name}...")
+    start_time = time()
+    projections[name] = transformer.fit_transform(data, y)
+    timing[name] = time() - start_time
+
+# %%
+# Finally, we can plot the resulting projection given by each method.
+from itertools import zip_longest
+
+fig, axs = plt.subplots(nrows=7, ncols=2, figsize=(15, 30))
```

Comment:
To make the figures look a little better and fill up the space:

```python
fig, axs = plt.subplots(nrows=7, ncols=2, figsize=(12, 24))

...

fig.tight_layout()
plt.show()
```

Using `zip_longest` longest here is nice. :)

Reply:
Thanks I was indeed searching for the best sizing :)